fairseq/training.py

```diff
@@ -5,56 +5,117 @@
 import sys
 import copy
 import re
+from typing import Dict, Any
 
-from sisyphus import *
+from sisyphus import Task, Job, tk
+from sisyphus.hash import sis_hash_helper
 
-import recipe.i6_core.util as util
+import i6_core.util as util
 
 
 class FairseqHydraConfig:
     """
-    An object that manages a Fairseq hydra config (inspired by the ReturnnConfig).
+    An object that manages a Fairseq hydra config.
     """
 
-    def __init__(self, config_dict, *, yaml_prefix=""):
+    def __init__(
+        self,
+        config_dict: Dict[str, Any],
+        post_config_dict: Dict[str, Any],
+        *,
+        package_name: str = "",
+    ):
         """
-        :param dict config_dict: Contains the information which is needed for fairseq-hydra-train. Will be converted and dumped into a .yaml
-        :param str yaml_prefix: Prefix which should be written to the beginning of the config, for example "# @package _group_"
+        :param config_dict: Contains the information which is needed for fairseq-hydra-train. Will be converted and dumped into a .yaml
+        :param dict post_config_dict: dictionary of the FairseqHydraConfig config variables that are not hashed
+        :param package_name: The @package directory that is required to be added to the top of Hydra config, for example "# @package _group_"
         """
         assert isinstance(config_dict, dict)
+        assert isinstance(post_config_dict, dict)
         self.config_dict = config_dict
-        self.yaml_prefix = yaml_prefix
+        self.post_config_dict = post_config_dict if post_config_dict is not None else {}
+        self.package_name = package_name
+        self.check_consistency()
+
+    @staticmethod
+    def update_nested_dict(dict1, dict2):
+        for g in dict2:
+            if g in dict1:
+                dict1[g].update(dict2[g])
```

Comment:
The wav2vec 2 model only has maximum depth of 2 for the nested dict. But in case of the future usage for other models, it would be safer if we also consider nested dict with maximum depth of more than 2.  I have updated this function